src/renderer/dx/DxRenderer.cpp

```diff
@@ -560,6 +687,8 @@ try
         CATCH_LOG(); // A failure in the notification function isn't a failure to prepare, so just log it and go on.
     }
 
+    _recreateDeviceRequested = false;

```

Comment:
vaguely concerned about the implications of us NOT having this line earleir?

Reply:
woah were we... requesting the resources get created _every frame_?

Oh no, this moved here from L1121 in the original DXRenderer. There's not a real difference here - before if this function failed, we'd `RETURN_IF_FAILED` out of `StartPaint` before clearing the `_recreateDeviceResources`. Same thing happens now.